sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -1184,10 +1184,10 @@ def _create_venv_from_scratch(
         subprocess.run([venv_pip, 'install'] + packages, check=True)
         with open(venv + '-requirements.txt', 'w') as fout:
           fout.write('\n'.join(packages))
-      except:  # pylint: disable=bare-except
+      except Exception as e:
+        logging.warning("Failed to create venv %s from scratch: %s", venv, e, exc_info=True)
```

Comment:
_create_venv_from_scratch will return a non-existing path, and will fail elsewhere, even more difficult to debug if it happens.

I think no more action is needed. The dependency is added after 34073